docs/tutorials/client-side-encryption.txt

```diff
@@ -0,0 +1,178 @@
+including server administrators, cannot read the encrypted data.
+
+Installation
+------------
+
+Client-side encryption requires the installation of additional packages.
+
+libmongocrypt
+~~~~~~~~~~~~~
+
+Libmongocrypt is a C library used by the driver for client-side encryption.
+To use client-side encryption, you must install the libmongocrypt binary
+on the machine running your Ruby program.
+
+To download a pre-built binary:
+
+- Download a tarball of all libmongocrypt variations from this `AWS bucket <https://s3.amazonaws.com/mciuploads/libmongocrypt/all/master/latest/libmongocrypt-all.tar.gz>`_.
+
+- Unzip the file you downloaded. You will see a list of folders, each
+  corresponding to an operating system. Find the folder that matches your
+  operating system and open it.
+
+- Inside that folder, open the folder called "nocrypto." In either the
+  lib or lb64 folder, you will find the libmongocrypt.so or
+  libmongocrypt.dylib or libmongocrypt.dll file, depending on your OS.
+
+- Move that file to wherever you want to keep it on your machine. You may delete
+  the other files included in the tarball.
+
+To build binary from source:
+
+- Follow the instructions in the README on the `libmongocrypt GitHub repo <https://github.com/mongodb/libmongocrypt>`_.
+
+Once you have the libmongocrypt binary on your machine, specify the path to the
+binary using the LIBMONGOCRYPT_PATH environment variable. It is recommended that
+you add this variable to your rc files. For example:
+
+.. code-block:: bash
+
+  export LIBMONGOCRYPT_PATH=/path/to/your/libmongocrypt.so
+
+mongocryptd
+~~~~~~~~~~~
+
+Mongocryptd is a daemon that tells the driver which fields to encrypt in a
+given operation. It is only required for automatic encryption, which is an
+enterprise-only feature. If you only intend to use explicit encryption, you may
+skip this step.
+
+Mongocryptd comes pre-packaged with enterprise builds of the MongoDB server
+(versions 4.2 and newer). If you must install mongocryptd separately, follow
+the `installation instructions in the MongoDB manual <https://docs.mongodb.com/manual/reference/security-client-side-encryption-appendix/#installation>`_.
+
+Automatic Encryption
+--------------------
+
+Automatic encryption is a feature that allows users to configure a
+``Mongo::Client`` instance to always encrypt specific document fields when
+performing database operations. Once the ``Mongo::Client`` is configured, it
+will automatically encrypt any field that requires encryption before writing
+it to the database, and it will automatically decrypt those fields when reading
+them. Automatic encryption is an enterprise-only feature.
+
+The following example provides a demonstration of auto-encryption using a local
+master key.
+
+.. code-block:: ruby
+
+  require 'mongo'
+
+  # Generate a local encryption master key
+  local_master_key = Base64.encode64(SecureRandom.random_bytes(96))
```

Comment:
IRL, you wouldn't just generate a new master key like this, but I wanted to create a snippet of code that somebody can just copy and paste that actually works. I was thinking of making it clear in the "Creating a Master Key" section that you should generate the master key and then store it somewhere. Open to thoughts on this.

Reply:
This sounds good, for example adding a one-sentence comment suggesting persisting `local_master_key` somewhere is fine with me.